### ISSUE TITLE
OvenPlayer: Can not play due to unknown reasons

### ISSUE THREAD

**Issue by ds88888888** [AirenSoft/OvenMediaEngine#553]
I have successfully set up OME with OBS streaming via RTMP on a local server: rtmp://localhost:1935/app/stream. I am able to use the test player locally to confirm WebRTC playback with ws://<serverIP>/app/stream. 

However when I try to use a different PC on the same network, the test player is not able to play the stream. Am I missing some configuration here? I thought maybe something with a public STUN server, but as I am on the same network, I assume it shouldn't be an issue. All port numbers used are at their defaults in this setup.

**Comment by getroot** [comment]
It could be a CORS error in your web browser, or it could be a Mixed Content problem. Check the log in the developer console of your web browser.

Or your server's firewall may be blocking the port used by OME. Check your firewall.

It is difficult to know exactly because there is no information given other than this.

**Comment by ds88888888** [comment]
> It could be a CORS error in your web browser, or it could be a Mixed Content problem. Check the log in the developer console of your web browser.
> 
> Or your server's firewall may be blocking the port used by OME. Check your firewall.
> 
> It is difficult to know exactly because there is no information given other than this.

Thanks, appreciate that, I'll take a look. I made sure to open the firewall ports as per the Getting Started guide. Should that be enough? 

I am using Firefox on Ubuntu 20.04.3. Are there any general logs created by OME that I can also review? If so, where are they located? Anything else I can provide to help? Thanks.

**Comment by ds88888888** [comment]
There are no entries in ovenmediaengine.log, and no error messages when doing:
`$ systemctl status ovenmediaengine`

<pre><font color="#4E9A06">
<font color="#4E9A06"><b>●</b></font> ovenmediaengine.service - OvenMediaEngine
     Loaded: loaded (/lib/systemd/system/ovenmediaengine.service; enabled; vendor preset: enabled)
     Active: <font color="#4E9A06"><b>active (running)</b></font> since Thu 2021-11-11 08:24:07 GMT; 19min ago
    Process: 4977 ExecStart=/usr/bin/OvenMediaEngine -d (code=exited, status=0/SUCCESS)
   Main PID: 4979 (OvenMediaEngine)
      Tasks: 55 (limit: 4566)
     Memory: 60.1M
     CGroup: /system.slice/ovenmediaengine.service
             └─4979 /usr/bin/OvenMediaEngine -d

Nov 11 08:24:07 linuxbox systemd[1]: Starting OvenMediaEngine...
Nov 11 08:24:07 linuxbox systemd[1]: Started OvenMediaEngine.
</pre><br>


There are also no changes made to Server.xml and everything is at the defaults. I have also noticed that have used rtmp://localhost... in OBS, if it mattered at all, I've changed it to the local IP address of the server and I'm still getting issues playing on a different PC that is on the same network. Browser reports it is latest version of Chrome (95.0.4638.69).


When opening ports with `firewall-cmd` it returned `success` - confirmed by doing:
`$ firewall-cmd --list-ports`

<pre>80/tcp 1935/tcp 9999/udp 4000-4005/udp 3333/tcp 3478/tcp 9000/tcp 10000-10010/udp
</pre>

Anything else to check?

**Comment by getroot** [comment]
Logs are located in /var/log/ovenmediaenigne/ . Please refer to the manual.

https://airensoft.gitbook.io/ovenmediaengine/logs-and-statistics#logs

Open the developer tools of Firefox or Chrome and look at the console log. Aren't there any Mixed Contents warnings?
Are you using ws://xxx:3333/aaa/bbb/ from https://demo.ovenplayer.com? Loading ws from https gives Mixed Content error, so you need to load stream from http://demo.ovenplayer.com. (This is allowed except for local addresses.)

**Comment by ds88888888** [comment]
> Open the developer tools of Firefox or Chrome and look at the console log. Aren't there any Mixed Contents warnings? Are you using ws://xxx:3333/aaa/bbb/ from https://demo.ovenplayer.com? Loading ws from https gives Mixed Content error, so you need to load stream from http://demo.ovenplayer.com. (This is allowed except for local addresses.)

Sorry I forgot to check that! Here is a screenshot of when I enter `ws://<server internal IP>:3333/app/stream` with protocol as WebRTC selected at http://demo.ovenplayer.com/ on the second PC on the same network.

![OME](https://user-images.githubusercontent.com/93290802/141298313-c4e11433-e84f-47f8-8e15-8ca8bf8d2bd7.PNG)

There are no new events in the log, only when I do the same process on the server locally, using http test player and ws://.

I have not installed TLS certificate to OME but I assume that's only if I want to use https:// and wss://.

Would you recommend anything else to try?

**Comment by getroot** [comment]
According to the screen you captured, the WebSocket connection fails.

The TCP connection from your Second PC to the port <server internal IP>:3333 is failing before it even reaches OvenMediaEngine.

This is a network error, and it doesn't seem to be in the range I could expect anymore. Find out why your Second PC cannot connect to <server internal IP>:3333/tcp. I think your network configuration or settings may be wrong.

**Comment by ds88888888** [comment]
> According to the screen you captured, the WebSocket connection fails.
> 
> The TCP connection from your Second PC to the port `<server internal IP>:3333` is failing before it even reaches OvenMediaEngine.
> 
> This is a network error, and it doesn't seem to be in the range I could expect anymore. Find out why your Second PC cannot connect to `<server internal IP>:3333/tcp`. I think your network configuration or settings may be wrong. (Or did you misspelled `<server internal IP>`?)

In the current configuration, I am able to ping the gateway from both devices, and each other. I have double checked the server IP is correct when trying to use the test player. 

I will report back after checking what else in the network may be causing problems. I am wondering could it be that a managed switch in the middle of it all could be causing issues? There shouldn't be any restrictions on those ports anyway. I'll take it out of the equation and test with just server, PC and a router connected to the internet.

**Comment by getroot** [comment]
Please share the results of the following commands. You seem to be using a private IP, so you don't have to hide your IP, isn't it?

```
$ ip addr

$ sudo netstat -lnpt | grep 3333

$ sudo ufw status
```

**Comment by ds88888888** [comment]
> Please share the results of the following commands. You seem to be using a private IP, so you don't have to hide your IP, isn't it?
> 
> ```
> $ ip addr
> 
> $ sudo netstat -lnpt | grep 3333
> 
> $ sudo ufw status
> 
> $ sudo iptables -L -n
> ```

I paste the outputs of all below. I notice that for the third one, ufw status is inactive. Does that mean that all the commands I issued previously as per the getting started guide with firewalld has not taken effect?

<pre><font color="#4E9A06"><b>superstonk@linuxbox</b></font>:<font color="#3465A4"><b>~</b></font>$ ip addr
1: lo: &lt;LOOPBACK,UP,LOWER_UP&gt; mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp16s0: &lt;BROADCAST,MULTICAST,UP,LOWER_UP&gt; mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 10:c3:7b:47:4d:56 brd ff:ff:ff:ff:ff:ff
    inet 10.80.25.103/24 brd 10.80.25.255 scope global noprefixroute enp16s0
       valid_lft forever preferred_lft forever
    inet6 fe80::a863:70c5:6ad:54ba/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
3: enp17s0: &lt;NO-CARRIER,BROADCAST,MULTICAST,UP&gt; mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether 10:c3:7b:47:4d:57 brd ff:ff:ff:ff:ff:ff
4: docker0: &lt;NO-CARRIER,BROADCAST,MULTICAST,UP&gt; mtu 1500 qdisc noqueue state DOWN group default 
    link/ether 02:42:4b:40:39:71 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
</pre>

<pre><font color="#4E9A06"><b>superstonk@linuxbox</b></font>:<font color="#3465A4"><b>~</b></font>$ sudo netstat -lnpt | grep 3333
[sudo] password for streambox: 
tcp        0      0 0.0.0.0:<font color="#CC0000"><b>3333</b></font>            0.0.0.0:*               LISTEN      3929/OvenMediaEngin 
</pre>

<pre><font color="#4E9A06"><b>superstonk@linuxbox</b></font>:<font color="#3465A4"><b>~</b></font>$ sudo ufw status
Status: inactive
</pre>

<pre><font color="#4E9A06"><b>superstonk@linuxbox</b></font>:<font color="#3465A4"><b>~</b></font>$ sudo iptables -L -n
Chain INPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED,DNAT
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
INPUT_direct  all  --  0.0.0.0/0            0.0.0.0/0           
INPUT_ZONES  all  --  0.0.0.0/0            0.0.0.0/0           
DROP       all  --  0.0.0.0/0            0.0.0.0/0            ctstate INVALID
REJECT     all  --  0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited

Chain FORWARD (policy DROP)
target     prot opt source               destination         
DOCKER-USER  all  --  0.0.0.0/0            0.0.0.0/0           
DOCKER-ISOLATION-STAGE-1  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
DOCKER     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED,DNAT
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
FORWARD_direct  all  --  0.0.0.0/0            0.0.0.0/0           
FORWARD_IN_ZONES  all  --  0.0.0.0/0            0.0.0.0/0           
FORWARD_OUT_ZONES  all  --  0.0.0.0/0            0.0.0.0/0           
DROP       all  --  0.0.0.0/0            0.0.0.0/0            ctstate INVALID
REJECT     all  --  0.0.0.0/0            0.0.0.0/0            reject-with icmp-host-prohibited

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           
OUTPUT_direct  all  --  0.0.0.0/0            0.0.0.0/0           

Chain DOCKER (1 references)
target     prot opt source               destination         

Chain DOCKER-ISOLATION-STAGE-1 (1 references)
target     prot opt source               destination         
DOCKER-ISOLATION-STAGE-2  all  --  0.0.0.0/0            0.0.0.0/0           
RETURN     all  --  0.0.0.0/0            0.0.0.0/0           

Chain DOCKER-ISOLATION-STAGE-2 (1 references)
target     prot opt source               destination         
DROP       all  --  0.0.0.0/0            0.0.0.0/0           
RETURN     all  --  0.0.0.0/0            0.0.0.0/0           

Chain DOCKER-USER (1 references)
target     prot opt source               destination         
RETURN     all  --  0.0.0.0/0            0.0.0.0/0           

Chain FORWARD_IN_ZONES (1 references)
target     prot opt source               destination         
FWDI_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
FWDI_docker  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
FWDI_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 

Chain FORWARD_OUT_ZONES (1 references)
target     prot opt source               destination         
FWDO_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
FWDO_docker  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
FWDO_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 

Chain FORWARD_direct (1 references)
target     prot opt source               destination         

Chain FWDI_docker (1 references)
target     prot opt source               destination         
FWDI_docker_pre  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_docker_log  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_docker_deny  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_docker_allow  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_docker_post  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           

Chain FWDI_docker_allow (1 references)
target     prot opt source               destination         

Chain FWDI_docker_deny (1 references)
target     prot opt source               destination         

Chain FWDI_docker_log (1 references)
target     prot opt source               destination         

Chain FWDI_docker_post (1 references)
target     prot opt source               destination         

Chain FWDI_docker_pre (1 references)
target     prot opt source               destination         

Chain FWDI_public (2 references)
target     prot opt source               destination         
FWDI_public_pre  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_public_log  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_public_deny  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_public_allow  all  --  0.0.0.0/0            0.0.0.0/0           
FWDI_public_post  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0           

Chain FWDI_public_allow (1 references)
target     prot opt source               destination         

Chain FWDI_public_deny (1 references)
target     prot opt source               destination         

Chain FWDI_public_log (1 references)
target     prot opt source               destination         

Chain FWDI_public_post (1 references)
target     prot opt source               destination         

Chain FWDI_public_pre (1 references)
target     prot opt source               destination         

Chain FWDO_docker (1 references)
target     prot opt source               destination         
FWDO_docker_pre  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_docker_log  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_docker_deny  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_docker_allow  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_docker_post  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           

Chain FWDO_docker_allow (1 references)
target     prot opt source               destination         

Chain FWDO_docker_deny (1 references)
target     prot opt source               destination         

Chain FWDO_docker_log (1 references)
target     prot opt source               destination         

Chain FWDO_docker_post (1 references)
target     prot opt source               destination         

Chain FWDO_docker_pre (1 references)
target     prot opt source               destination         

Chain FWDO_public (2 references)
target     prot opt source               destination         
FWDO_public_pre  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_public_log  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_public_deny  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_public_allow  all  --  0.0.0.0/0            0.0.0.0/0           
FWDO_public_post  all  --  0.0.0.0/0            0.0.0.0/0           

Chain FWDO_public_allow (1 references)
target     prot opt source               destination         

Chain FWDO_public_deny (1 references)
target     prot opt source               destination         

Chain FWDO_public_log (1 references)
target     prot opt source               destination         

Chain FWDO_public_post (1 references)
target     prot opt source               destination         

Chain FWDO_public_pre (1 references)
target     prot opt source               destination         

Chain INPUT_ZONES (1 references)
target     prot opt source               destination         
IN_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
IN_docker  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 
IN_public  all  --  0.0.0.0/0            0.0.0.0/0           [goto] 

Chain INPUT_direct (1 references)
target     prot opt source               destination         

Chain IN_docker (1 references)
target     prot opt source               destination         
IN_docker_pre  all  --  0.0.0.0/0            0.0.0.0/0           
IN_docker_log  all  --  0.0.0.0/0            0.0.0.0/0           
IN_docker_deny  all  --  0.0.0.0/0            0.0.0.0/0           
IN_docker_allow  all  --  0.0.0.0/0            0.0.0.0/0           
IN_docker_post  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     all  --  0.0.0.0/0            0.0.0.0/0           

Chain IN_docker_allow (1 references)
target     prot opt source               destination         

Chain IN_docker_deny (1 references)
target     prot opt source               destination         

Chain IN_docker_log (1 references)
target     prot opt source               destination         

Chain IN_docker_post (1 references)
target     prot opt source               destination         

Chain IN_docker_pre (1 references)
target     prot opt source               destination         

Chain IN_public (2 references)
target     prot opt source               destination         
IN_public_pre  all  --  0.0.0.0/0            0.0.0.0/0           
IN_public_log  all  --  0.0.0.0/0            0.0.0.0/0           
IN_public_deny  all  --  0.0.0.0/0            0.0.0.0/0           
IN_public_allow  all  --  0.0.0.0/0            0.0.0.0/0           
IN_public_post  all  --  0.0.0.0/0            0.0.0.0/0           
ACCEPT     icmp --  0.0.0.0/0            0.0.0.0/0           

Chain IN_public_allow (1 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:22 ctstate NEW,UNTRACKED

Chain IN_public_deny (1 references)
target     prot opt source               destination         

Chain IN_public_log (1 references)
target     prot opt source               destination         

Chain IN_public_post (1 references)
target     prot opt source               destination         

Chain IN_public_pre (1 references)
target     prot opt source               destination         

Chain OUTPUT_direct (1 references)
target     prot opt source               destination</pre>

**Comment by ds88888888** [comment]
I had also just realised that by doing `firewall-cmd --list-ports` the list was empty. Do the changes not save after a reboot? I had to add all the ports again as follows;

```
$ sudo firewall-cmd --add-port=80/tcp
$ sudo firewall-cmd --add-port=1935/tcp
$ sudo firewall-cmd --add-port=9999/udp
$ sudo firewall-cmd --add-port=4000-4005/udp
$ sudo firewall-cmd --add-port=3333/tcp
$ sudo firewall-cmd --add-port=3478/tcp
$ sudo firewall-cmd --add-port=9000/tcp
$ sudo firewall-cmd --add-port=10000-10010/udp
```
And obviously the output of `sudo iptables -L -n` has updated to reflect the above change to add the relevant ports;

<pre>Chain IN_public_allow (1 references)
target     prot opt source               destination         
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:22 ctstate NEW,UNTRACKED
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:80 ctstate NEW,UNTRACKED
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:1935 ctstate NEW,UNTRACKED
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:9999 ctstate NEW,UNTRACKED
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpts:4000:4005 ctstate NEW,UNTRACKED
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:3333 ctstate NEW,UNTRACKED
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:3478 ctstate NEW,UNTRACKED
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:9000 ctstate NEW,UNTRACKED
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpts:10000:10010 ctstate NEW,UNTRACKED
</pre>

**Comment by ds88888888** [comment]
Sorry about the multiple posts, I also have a new error when doing `sudo systemctl status ovenmediengine.sevice`

<pre>Nov 15 09:19:19 streambox OvenMediaEngine[3010]: [2021-11-15 09:19:19.093] W [SPSegPub-T80:3020] HttpServer | http_server.cpp:223  | No module could be found to handle this connection request : [ws://10.80.25.103/app/stream]<pre>

**Comment by getroot** [comment]
ws://10.80.25.103/app/stream

You didn't specify port 3333.

**Comment by ds88888888** [comment]
> ws://10.80.25.103/app/stream
> 
> You didn't specify port 3333.

Added port 3333 to the URL and I am able to view the stream on the second PC. I'm certain that I did this full URL last week and it didn't work. What I suspect what may have happened last time, was that I rebooted the server assuming the open ports done with `firewalld` were permanent. This isn't done by default for some reason and I'm not sure why? Thank you so much for your help, I really appreciate it.
